compiler/rustc_feature/src/active.rs

```diff
@@ -671,6 +671,9 @@ declare_features! (
     /// Allows unnamed fields of struct and union type
     (active, unnamed_fields, "1.53.0", Some(49804), None),
 
+    /// Allows struct-structs with qualified paths in patterns
+    (active, qualified_path_in_patterns, "1.54.0", Some(80080), None),
```

Comment:
qpath in struct-struct expressions already exist in the language so this would be a breaking change. 

qpath in tuple-struct expressions and patterns still are not supported. Should we still put them behind the gate even though they're still not possible to use?

Reply:
(re-reading your previous comments, you indicate that you think tuple-struct expressions/patterns should work so I agree, if I can get those working, we should also put them behind the feature flag)